skrub/_reporting/tests/test_table_report.py

```diff
@@ -123,6 +126,49 @@ def test_duration(df_module):
     assert re.search(r"2(\.0)?\s+days", TableReport(df).html())
 
 
+@pytest.mark.parametrize(
+    "filename_type",
+    ["str", "Path", "file_object", "binary_mode"],
+)
+def test_write_html(tmp_path, pd_module, filename_type):
```

Comment:
this test is looking great! the last thing we need to take care of is to make sure we close the file if we opened it (otherwise we "leak" a resource: the opened file handle. that could be a problem for example to clean up the temp directory on windows as it refuses to remove files that have an open file handle). Usually we ensure that with a simple context manager like:

```python
with open(tmp_file_path, 'w', encoding='utf-8') as file:
    file.write('hello')
```

but here we have a tricky situation because in some cases we have a string or path (which require no closing) and sometimes we have a file object which does require closing.

The standard library module `contextlib` provides 2 ways to deal with that situation easily. The first is `ExitStack`: it creates a context and we can push as many context managers as we want to its stack; when it exits it unwinds the stack, calling each manager's `__exit__` when it is popped. So we could use it like:

```python
with contextlib.ExitStack() as stack:
    if file_type == 'str':
        file = str(tmp_file_path)
    elif file_type == 'text_file_object':
        file = stack.enter_context(open(tmp_file_path, 'w', encoding='utf-8'))
    # ...

    report.write_html(file)

# if we opened it the file is closed here when we exit the `with` block
```

This option using ExitStack is my favorite because the file is being managed by a context manager as soon as it is opened.

Another way is to use `nullcontext` in the cases where we do not open the file, so that later we can treat all options as if they were open files that implement the context manager protocol. `nullcontext` returns an object that implements the context manager protocol but whose `__enter__` just returns the object we gave it and `__exit__` does nothing:

```python
if file_type == 'str':
    file = contextlib.nullcontext(str(tmp_file_path))
elif file_type == 'text_file_object':
    file = open(tmp_file_path, 'w', encoding='utf-8')

with file:
    report.write_html(file)
```

Reply:
wow contextlib.ExitStack() is very nice

Reply:
Maybe you could add a comment on L139 to explain why it's a good idea to use contextlib here?